src/goTest.ts

```diff
@@ -153,17 +151,19 @@ export function testCurrentFile(goConfig: vscode.WorkspaceConfiguration, args: s
 		return;
 	}
 
-	return getTestFunctions(editor.document).then(testFunctions => {
-		const testConfig = {
-			goConfig: goConfig,
-			dir: path.dirname(editor.document.fileName),
-			flags: getTestFlags(goConfig, args),
-			functions: testFunctions.map(func => { return func.name; })
-		};
-		// Remember this config as the last executed test.
-		lastTestConfig = testConfig;
+	editor.document.save().then(() => {
```

Comment:
The tests are failing due to the missing `return` here.
The `testCurrentFile` method should return a promise